server/network/aoprotocol.py

```diff
@@ -430,8 +430,10 @@ def net_cmd_ms(self, args):
             return
         if sfx_delay < 0:
             return
-        if button not in (0, 1, 2, 3, 4):
-            return
+        if '4' in button and "<and>" not in button:
```

Comment:
Could you explain this logic?

Reply:
Yes, it checks whether the packet is valid. e.g:
"4<and>abc.gif" -> valid as both 4 and <and> exist
"4abcdef" -> invalid format as 4 exist but "<and>" doesn't
"4" -> valid as its a custom objection